File: src/switchnobreaks/Switch.java
package switchnobreaks;

import org.jetbrains.annotations.Nullable;

/**
 * @author Sergey Ryabov <colriot@gmail.com>
 *         20/06/2017
 */
public class Switch {

  private static String describe(@Nullable Integer i) {
    if (i == null) return "null";

    switch (i) {
      case 1: case 2: case 3: case 4:
        return "less than FIVE";

      case 5:
        return "FIVE";

      default:
        return "Error: Unexpected number!";
    }
  }

  public static void main(String[] args) {
    System.out.println(describe(42));
  }
}

File: src/switchnobreaks/Switch.kt
package switchnobreaks

/**
 * @author Sergey Ryabov <colriot@gmail.com>
 * 20/06/2017
 */
private fun describe(i: Number?) =
    when (i) {
      null        -> "null"
      in 1..4, -3 -> "less than FIVE"
      5           -> "FIVE"
      is Double   -> "WOW Double"
      !in 1..100  -> {
        "not in FIRST HUNDRED"
      }
      else        -> {
        "Error: Unexpected number!"
      }
    }

// Just a list of conditions to be tested.
fun advanced(x: Int) =
    when {
      x.isOdd()  -> print("x is odd")
      x.isEven() -> print("x is even")
      else       -> print("x is funny")
    }

private fun Int.isOdd() = this % 2 != 0
private fun Int.isEven() = !this.isOdd()

fun main(args: Array<String>) {
  println(describe(42.0))
  advanced(42)
}
